my-vue3-app/src/store/reports/actions.ts

```diff
@@ -1,21 +1,56 @@
 import { ActionContext } from 'vuex';
 import api from '@/services/api';
-import { Pagination } from './state';
-import { SET_DATA, SET_PAGINATION, SET_LOADING } from './mutations';
+import {
+  PAGINATION, Pagination, Reports, REPORTS, State,
+} from './state';
+import {
+  ADD_DATA, SET_PAGINATION, SET_LOADING,
+} from './mutations';
 
 export const FETCH_REPORTS = 'load_reports';
 
+const applyFilterSortToReports = (
+  pagination: Pick<Pagination, 'order'>,
+  reports: Reports,
+): Array<string> => Object
+  .keys(reports)
+  // .map((id) => reports[id]);
+  // filter
+  // .map((report) => report.id);
+  .sort((a, b) => a.localeCompare(b) * pagination.order.id);
+
```

Comment:
why we need this sort?

Reply:
to make linear extrapolation we need linear function (index, index + 1, index + 2 ...) it can be true if only local order is the same as remote order.
we have object with reports, NOT an ARRAY. Objects are supposed to not have strict order (I may be wrong, but there is no ability to manipulate order of the object).

Reply:
Some details https://2ality.com/2015/10/property-traversal-order-es6.html
You can change order https://stackoverflow.com/questions/6959817/changing-the-order-of-the-object-keys but it is strange for me.
anyway, first statement ha

Reply:
The order is strange :)
![image](https://user-images.githubusercontent.com/5002775/142920774-da2c3431-e0c4-4738-8481-c80b6b8af9f3.png)